src/changes/findChanges.ts

```diff
@@ -47,21 +47,6 @@ export function findChanges(state: EditorState) {
     for (let i = 0; i < tracked.length; i += 1) {
       const dataTracked = tracked[i]
       const id = dataTracked.id || ''
-      // Join adjacent text changes that have been broken up due to different marks
-      // eg <ins><b>bold</b>norm<i>italic</i></ins> -> treated as one continuous change
-      // Note the !equalMarks to leave changes separate incase the marks are equal to let fixInconsistentChanges to fix them
-      if (
-        current &&
-        current.change.id === id &&
-        current.node.isText &&
-        node.isText &&
-        !equalMarks(node, current.node)
-      ) {
-        current.change.to = pos + node.nodeSize
-        // Important to update the node as the text changes might contain multiple parts where some marks equal each other
-        current.node = node
-        continue
-      }
```

Comment:
I removed this as that will keep the same change Id for text change. and we are going to do the join from changeTree for that case